app/utils/postMessage.js

```diff
@@ -30,12 +30,12 @@ function _isTopLevelWindow() {
 /**
  * store for message handle callbacks
  */
-const _callbacks = {};
+export const _callbacks = {};
 
 /**
  * Setup postMessage receive callbacks
  */
-export function setupPostMessage() {
+export function setupPostMessage(window = window) {
```

Comment:
Adding `window` as optional parameter helps with testability by allowing us to pass in simulated window conditions. (Inspired by [this stack thread](http://stackoverflow.com/a/8922356).)